test/e2e/wait-for-nginx.sh

```diff
@@ -14,48 +14,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-echo "deploying NGINX Ingress controller"
-cat deploy/namespace.yaml | kubectl apply -f -
-cat deploy/default-backend.yaml | kubectl apply -f -
-cat deploy/configmap.yaml | kubectl apply -f -
-cat deploy/tcp-services-configmap.yaml | kubectl apply -f -
-cat deploy/udp-services-configmap.yaml | kubectl apply -f -
-cat deploy/without-rbac.yaml | kubectl apply -f -
-cat deploy/provider/baremetal/service-nodeport.yaml | kubectl apply -f -
+NAMESPACE=$1
 
-echo "updating image..."
-kubectl set image \
-    deployments \
-    --namespace ingress-nginx \
-	--selector app=ingress-nginx \
-    nginx-ingress-controller=quay.io/kubernetes-ingress-controller/nginx-ingress-controller:test
+echo "deploying NGINX Ingress controller in namespace $NAMESPACE"
 
-sleep 5
-
-echo "waiting NGINX ingress pod..."
-
-function waitForPod() {
-    until kubectl get pods -n ingress-nginx -l app=ingress-nginx -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True";
-    do
-        sleep 1;
-    done
-}
-
-export -f waitForPod
+cat $DIR/../manifests/ingress-controller/default-backend.yaml            | kubectl create --namespace=$NAMESPACE -f -
+cat $DIR/../manifests/ingress-controller/configmap.yaml                  | kubectl create --namespace=$NAMESPACE -f -
+cat $DIR/../manifests/ingress-controller/tcp-services-configmap.yaml     | kubectl create --namespace=$NAMESPACE -f -
+cat $DIR/../manifests/ingress-controller/udp-services-configmap.yaml     | kubectl create --namespace=$NAMESPACE -f -
+cat $DIR/../manifests/ingress-controller/with-rbac.yaml                  | kubectl create --namespace=$NAMESPACE -f -
+cat $DIR/../manifests/ingress-controller/service-nodeport.yaml           | kubectl create --namespace=$NAMESPACE -f -
 
-timeout 30s bash -c waitForPod
-
-if kubectl get pods -n ingress-nginx -l app=ingress-nginx -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True";
-then
-    echo "Kubernetes deployments started"
-else
-    echo "Kubernetes deployments with issues:"
-    kubectl get pods -n ingress-nginx
-
-    echo "Reason:"
-    kubectl describe pods -n ingress-nginx
-    kubectl logs -n ingress-nginx -l app=ingress-nginx
-    exit 1
-fi
+sleep 5
```

Comment:
Why did you decide to not keep the existing `waitForPod` approach? That seemed more resilient to me than hard coded 5s sleep.

Reply:
but that 30s is a timeout threshold, it does not necessarily wait 30s.

Reply:
> Also now we use the same wait logic than the rest of the tests

In the rest of the tests for ingress-nginx deployment we use `WaitForPodsReady`(i.e
https://github.com/kubernetes/ingress-nginx/pull/2374/files#diff-e427cee5cedcaaa18c6ac5414ea08183R379) which is more aligned with the previous implementation in this script IMHO

Reply:
done